webapp/src/components/emptyCenterPanel.tsx

```diff
@@ -1,34 +1,170 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
-import React from 'react'
-import {FormattedMessage} from 'react-intl'
+import React, {useCallback, useEffect} from 'react'
+import {FormattedMessage, useIntl} from 'react-intl'
+import {generatePath, useHistory, useRouteMatch} from 'react-router-dom'
 
 import {getCurrentWorkspace} from '../store/workspace'
-import {useAppSelector} from '../store/hooks'
+import {useAppSelector, useAppDispatch} from '../store/hooks'
+import {Utils} from '../utils'
+import {Board} from '../blocks/board'
+import {getGlobalTemplates, fetchGlobalTemplates} from '../store/globalTemplates'
+import {getSortedTemplates} from '../store/boards'
+import AddIcon from '../widgets/icons/add'
+import BoardIcon from '../widgets/icons/board'
+import octoClient from '../octoClient'
+
+import {addBoardTemplateClicked, addBoardClicked} from './sidebar/sidebarAddBoardMenu'
+import {addBoardFromTemplate, BoardTemplateMenuItemOptionsButton} from './sidebar/boardTemplateMenuItem'
+
 import './emptyCenterPanel.scss'
 
-const EmptyCenterPanel = React.memo(() => {
-    const workspace = useAppSelector(getCurrentWorkspace)
+type ButtonProps = {
+    buttonIcon: string | React.ReactNode,
+    title: string,
+    readonly: boolean,
+    onClick: () => void,
+    showBoard?: (boardId: string) => void
+    boardTemplate?: Board
+}
+
+const EmptyCenterPanelButton = React.memo((props: ButtonProps) => {
```

Comment:
Can we just call this a `PanelButton`? There isn't anything requiring it to only be used in the `EmptyCenterPanel` is there?

Reply:
Sure! Thanks. Done